cmd/kured/main.go

```diff
@@ -315,6 +325,32 @@ type nodeMeta struct {
 	Unschedulable bool `json:"unschedulable"`
 }
 
+func addNodeAnnotation(client *kubernetes.Clientset, nodeID, key, val string) {
+	node, err := client.CoreV1().Nodes().Get(context.TODO(), nodeID, metav1.GetOptions{})
+	if err != nil {
+		log.Fatal("Error retrieving node object via k8s API: %v", err)
+	}
+	node.Annotations[key] = val
+	log.Infof("Node %s Annotation: %s=%s", node.GetName(), key, val)
+	_, err = client.CoreV1().Nodes().Update(context.TODO(), node, metav1.UpdateOptions{})
```

Comment:
@devigned @CecileRobertMichon @mboersma are any of you aware of a patch-capable API call that can push a node annotation update (add/delete/update) via the k8s API? So that I don't have to add an additional ClusterRole node verb to the helm chart spec? (see the change to `charts/kured/templates/clusterrole.yaml` above)

Reply:
this discussion may be relevant to what you're looking for: https://github.com/kubernetes-sigs/cluster-api/pull/4048#discussion_r552794827

I ended up using the Cluster API patch helper in that PR so that it doesn't overwrite other changes.

Reply:
Yes, thank you, that is an excellent example. The main takeaway is that without using patch we are succeptible to a race condition where we conflict with another Update operation that happens between our original "get node object" call and the "update node" request.

I'll change to patch.